src/credential/sd_jwt.rs

```diff
@@ -0,0 +1,364 @@
+        };
+
+        // Check the JSON-encoded credential against the definition.
+        definition.check_credential_validation(&json)
+    }
+}
+
+#[uniffi::export]
+impl SdJwt {
+    /// Create a new SdJwt instance from a compact SD-JWS string.
+    #[uniffi::constructor]
+    pub fn new_from_compact_sd_jwt(input: String) -> Result<Arc<Self>, SdJwtError> {
+        let inner: SdJwtBuf =
+            SdJwtBuf::new(input).map_err(|e| SdJwtError::InvalidSdJwt(e.to_string()))?;
+
+        let mut sd_jwt = SdJwt::try_from(inner)?;
+        sd_jwt.key_alias = None;
+
+        Ok(Arc::new(sd_jwt))
+    }
+
+    /// Create a new SdJwt instance from a compact SD-JWS string with a provided key alias.
+    #[uniffi::constructor]
+    pub fn new_from_compact_sd_jwt_with_key(
+        input: String,
+        key_alias: KeyAlias,
+    ) -> Result<Arc<Self>, SdJwtError> {
+        let inner: SdJwtBuf =
+            SdJwtBuf::new(input).map_err(|e| SdJwtError::InvalidSdJwt(e.to_string()))?;
+
+        let mut sd_jwt = SdJwt::try_from(inner)?;
+        sd_jwt.key_alias = Some(key_alias);
+
+        Ok(Arc::new(sd_jwt))
+    }
+
+    /// Create an instance of a SdJwt from a generic credential type.
+    #[uniffi::constructor]
+    pub fn new_from_credential(credential: &Credential) -> Result<Arc<Self>, SdJwtError> {
+        Self::try_from(credential).map(Arc::new)
+    }
+
+    /// Return the ID for the SdJwt instance.
+    pub fn id(&self) -> Uuid {
+        self.id
+    }
+
+    /// Return the key alias for the credential
+    pub fn key_alias(&self) -> Option<KeyAlias> {
+        self.key_alias.clone()
+    }
+
+    /// The type of this credential. Note that if there is more than one type (i.e. `types()`
+    /// returns more than one value), then the types will be concatenated with a "+".
+    pub fn r#type(&self) -> CredentialType {
+        CredentialType(self.types().join("+"))
+    }
+
+    /// Decode a SdJwt instance and return the revealed claims as a JSON string.
+    pub fn decode_reveal_json_string(&self) -> Result<String, SdJwtError> {
+        serde_json::to_string(&self.revealed_claims)
+            .map_err(|e| SdJwtError::Serialization(e.to_string()))
+    }
+
+    /// Access the revealed SD-JWT disclosures as an vector of UTF-8 encoded strings.
+    pub fn disclosures(&self) -> Vec<String> {
+        self.inner
+            .as_sd_jwt()
+            .disclosures()
+            // TODO: Is there a better type to map to versus string?
+            // e.g., should we return a JSON object here?
+            .map(|disclosure| disclosure.as_str().to_string())
+            .collect()
+    }
+
+    /// Return the SdJwt as a JwtVc instance.
+    pub fn as_jwt_vc(self: Arc<Self>) -> Result<Arc<JwtVc>, SdJwtError> {
+        JwtVc::from_compact_jws_bytes(
+            self.id,
+            self.inner.as_bytes().into(),
+            self.key_alias.clone(),
+        )
+        .map_err(|e| SdJwtError::JwtVcInitError(e.to_string()))
+    }
```

Comment:
Why do we need to convert this to a different credential type?

Reply:
Removed this method completely.